Design/CrashReporting/DesignDocument.md

```diff
@@ -0,0 +1,179 @@
+
+##### Mantid Launcher
+
+The Mantid Launcher should be lightweight and invisible to the user. On Windows and Linux Mantid is already launched via a batch and bash script respectively which can be extended upon to fulfil this purpose. 
+
+Preliminary testing on Windows shows Mantid obeying the following exit code structure.
+
+Situation  | Exit Code   |  
+|--------|----------------|
+| Exited without error  | 0 | 
+| Force terminated  | 1 |
+| Crash to Desktop  | Varies but not so far 0 or 1|
+
+It should therefore be feasible to check the exit status of mantidplot and if necessary call the crash reporting implementation from within these scripts. A difficultly to keep in mind here however is that the crash reporting implementation in this instance needs to be called from outside Mantid.
+
+* Q This appears from preliminary testing to be feasible but needs to be checked on other possible configurations as well.
+
+### Crash Reporting
+The crash reporting will be implemented in the kernel in a similar way to the existing Usage service. This should allow it to be launchable from outside mantiplot. ConfigService may need to be modified to allow a skinny version to be launched here that does not launch all the algorithms. 
+
+This service should be exposed to the launcher through a python wrapper which can be accessed through mantidpython.
+
+The main things that this service needs to do are:
+* Check whether usage reporting is enabled and if not exit having done nothing.
+* Launch the user interface, via either the python wrapper or mantidplot.
+* Gather information on the crash
+* Send crash report to web service
+
+#### Information to gather
+
+##### Minimal information
+This is what will be gathered and sent if the mantiplot instance is not accessible and the user has opted to send no additional information.
+
+Field  | Data Type   |  
+|--------|----------------|
+| Crash Type  | TINYTEXT | 
+| Crash Time  | DATETIME |
+| Facility  | TINYTEXT|
+| Instrument  | TINYTEXT|
+| Mantid Version  | TINYTEXT|
+| Encrypted user ID  | TINYTEXT|
+| Encrypted host ID  | TINYTEXT|
+
+##### If mantidplot is open
+Field  | Data Type   |  
+|--------|----------------|
+| List of open Interfaces  | ENUM | 
+| Number of open graphs  | SMALLINT |
+| Number of loaded workspaces  | SMALLINT|
+
+##### If additional information is provided by user
+Field  | Data Type   |  
+|--------|----------------|
+| User name (if provided)  | TINYTEXT |
+| User email address (if provided)  | TINYTEXT | 
+| User phone number (if provided) | TINYTEXT |
+| User feedback (if provided)  | TEXT|
+| Last x characters of logfile  | TEXT|
+| Unencrypted user ID  | TINYTEXT|
+| Unencrypted host ID  | TINYTEXT|
+
+##### If additional information is provided by user and mantidplot is open
+Field  | Data Type   |  
+|--------|----------------|
+| MessageDialog text dump  | TEXT | 
```

Comment:
This is the information printed out within mantid plot in the MessageDisplay dialog box at the top of the window. As it is normally set at the notice level it may contain more information than what is printed out to the log file.

Reply:
The current defaults (and who changes defaults) is that everything (console, file, and result window) is logging at notice level.

Reply:
Oh ok if this is the case then this is no longer needed as it will just contain the exact same information as the log file.